src/net_processing.cpp

```diff
@@ -1926,6 +1926,20 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
         return true;
     }
 
+    // VeriBlock: if POP is not enabled, ignore POP-related P2P calls
+    if (VeriBlock::isPopActive()) {
+        LOCK(cs_main);
+        auto& state = *State(pfrom->GetId());
```

Comment:
I don't approve this because of LOCK(cs_main). 
We want to avoid locking cs_main for pop processing at any cost

Reply:
```
State* state = nullptr;
{
  LOCK(cs_main);
  state = State(pfrom->GetId());
}
```